proto/noble/dollar/vaults/v2/vaults.proto

```diff
@@ -0,0 +1,289 @@
+syntax = "proto3";
+
+package noble.dollar.vaults.v2;
+
+import "amino/amino.proto";
+import "cosmos_proto/cosmos.proto";
+import "gogoproto/gogo.proto";
+import "google/protobuf/timestamp.proto";
+import "noble/dollar/vaults/v1/vaults.proto";
+
+option go_package = "dollar.noble.xyz/v2/types/vaults/v2";
+
+// NOBLE DOLLAR V2 VAULT SYSTEM - YIELD-TRACKING ACCOUNTING
+//
+// This package defines the V2 vault system that uses direct yield tracking
+// for explicit yield accounting and improved transparency.
+//
+// Key features:
+// - Direct yield tracking per user position
+// - Explicit yield accrual accounting
+// - Exit queues for withdrawals
+// - User-controlled yield preferences
+// - Clean separation from legacy V1 system
+//
+// Migration approach: Users must withdraw from V1 and deposit into V2
+
+// UserPosition represents a user's position in the V2 yield-tracking vault system
+message UserPosition {
+  // Current deposit amount (principal)
+  string deposit_amount = 1 [
+    (cosmos_proto.scalar) = "cosmos.Int",
+    (gogoproto.customtype) = "cosmossdk.io/math.Int",
+    (gogoproto.nullable) = false,
+    (amino.dont_omitempty) = true
+  ];
+
+  // Accrued yield on the position
+  string accrued_yield = 2 [
+    (cosmos_proto.scalar) = "cosmos.Int",
+    (gogoproto.customtype) = "cosmossdk.io/math.Int",
+    (gogoproto.nullable) = false,
+    (amino.dont_omitempty) = true
+  ];
+
+  // Timestamp of first deposit
+  google.protobuf.Timestamp first_deposit_time = 3 [
+    (gogoproto.nullable) = false,
+    (gogoproto.stdtime) = true
+  ];
+
+  // Timestamp of last deposit/withdrawal
+  google.protobuf.Timestamp last_activity_time = 4 [
+    (gogoproto.nullable) = false,
+    (gogoproto.stdtime) = true
+  ];
+
+  // Whether user wants to receive yield (vs contributing to fee pool)
+  bool receive_yield = 5;
+
+  // Amount pending exit
+  string amount_pending_exit = 6 [
+    (cosmos_proto.scalar) = "cosmos.Int",
+    (gogoproto.customtype) = "cosmossdk.io/math.Int",
+    (gogoproto.nullable) = false,
+    (amino.dont_omitempty) = true
+  ];
+
+  // Number of active exit requests
+  int32 active_exit_requests = 7;
+}
+
+// ExitRequest represents a pending exit request
+message ExitRequest {
```

Comment:
`MsgRequestWithdrawal` and `ExitRequest` have inconsistent verbiage (exit vs withdraw).

Also, how does an `ExitReqeust` map to a user in state? I don't see a field that would be matched to the signer.